pkgs/by-name/ma/maven/package.nix

```
@@ -41,6 +44,15 @@ stdenvNoCC.mkDerivation (finalAttrs: {
     buildMavenPackage = callPackage ./build-maven-package.nix {
       maven = finalAttrs.finalPackage;
     };
+    mavenPatchPomHook =
+      makeSetupHook
+        {
+          name = "maven-patch-pom-hook";
+        }
+        (substituteAll {
+          src = ./hooks/patch-pom-hook.sh;
+          xmlstarlet = lib.getExe xmlstarlet;
+        });
```

I imagine `makeSetupHook` doesn't have the ability to add on to `PATH`?

---

I'm not sure. I've seen this pattern used twice, so I did something similar (but more override-friendly).

https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+%40xmlstarlet%40&type=code

---

Based on [this documentation section](https://github.com/NixOS/nixpkgs/blob/df8b26b3f795ed5eef2e98de88fe80efb949eb13/doc/build-helpers/special/makesetuphook.section.md?plain=1#L15-L55) the substitution has a purpose, but could be done in a cleaner way. Hance:

```suggestion
    mavenPatchPomHook =
      makeSetupHook
        {
          name = "maven-patch-pom-hook";
          propogatedBuildInputs = [ xmlstartlet ];
        }
        ./hooks/patch-pom-hook.sh;
```

(As well as editing the hook file itself.)

---

Neat.